### ISSUE TITLE
Parity node falls behind rest of network occasionally

### ISSUE THREAD

**Issue by russdirks** [ethcore/parity#3879]
My Parity node falls behind the rest of the network fairly often.  I'm on the ETC chain.  I wrote a little program to monitor, and on Dec 15, the results were as follows:

```
d15 03:25:55> The local node fell behind and has now caught up. It was behind 25.12 minutes / 102 blocks.  Current block = 2817599
d15 06:08:32> The local node fell behind and has now caught up. It was behind 16.54 minutes / 71 blocks.  Current block = 2818357
d15 09:02:27> The local node fell behind and has now caught up. It was behind 4.00 minutes / 28 blocks.  Current block = 2819139
d15 10:46:37> The local node fell behind and has now caught up. It was behind 4.00 minutes / 24 blocks.  Current block = 2819599
d15 11:23:12> The local node fell behind and has now caught up. It was behind 10.53 minutes / 44 blocks.  Current block = 2819744
d15 13:41:57> The local node fell behind and has now caught up. It was behind 55.11 minutes / 251 blocks.  Current block = 2820352
d15 19:36:11> The local node fell behind and has now caught up. It was behind 28.56 minutes / 110 blocks.  Current block = 2821893
d15 20:39:50> The local node fell behind and has now caught up. It was behind 20.08 minutes / 116 blocks.  Current block = 2822182
d15 23:07:54> The local node fell behind and has now caught up. It was behind 7.18 minutes / 46 blocks.  Current block = 2822850

```
Interestingly, I find that if I shut the node down and then restart it, it immediately catches up.  The first time I thought it was just a coincidence, but I've been able to do that at least 10 times ... it is very consistent.

Attached is a log file with the `-l sync=trace` option, as @arkpar instructed me on gitter.  I managed to capture a sync incident from around 8:42 to 8:58  (block 2825417 to 2825483).  If you want me to continue logging to capture more data, please let me know.
[parity_log.zip](https://github.com/ethcore/parity/files/657894/parity_log.zip)

**Comment by arkpar** [comment]
@russdirks Please try upgrading to 1.4.6 and see if it solves the issue

**Comment by russdirks** [comment]
Ok, upgraded to 1.4.6, and got a syncing incident around 14:40:00 (block 2827015) approximately.  I'll keep running with logging enabled to see if I can get another one.

**Comment by russdirks** [comment]
Sorry, forgot to upload the log.
[parity.zip](https://github.com/ethcore/parity/files/658479/parity.zip)

**Comment by russdirks** [comment]
Here's another one.  The sync incident starts at about 16:59:05, block 2827584.

[parity2.zip](https://github.com/ethcore/parity/files/658600/parity2.zip)

**Comment by Gustav-Simonsson** [comment]
Was going to create a new issue but perhaps its related: my parity node appears to have suddenly lost peers at block 2839407.
```
Parity/v1.4.6-beta-5feccf1-20161205/x86_64-linux-gnu/rustc1.12.0
./target/release/parity --port 30304 --cache-size-db 1024 --rpcport 8555
Ubuntu 16.04.1 LTS 4.4.0-36-generic
Non-virtualized, dedicated machine. beefy hardware.
```
Log:
```
Imported #2839403 3861<E2><80><A6>655f (20 txs, 0.49 Mgas, 3.31 ms, 3.39 KiB)
2016-12-19 13:44:57  Imported #2839404 7ca2<E2><80><A6>5466 (21 txs, 0.44 Mgas, 3.42 ms, 2.84 KiB)
2016-12-19 13:45:06      0/25/25 peers     23 MiB db    5 MiB chain  0 bytes queue  227 KiB sync
2016-12-19 13:45:19  Imported #2839405 975d<E2><80><A6>fe0c (7 txs, 0.15 Mgas, 5.52 ms, 1.30 KiB)
2016-12-19 13:45:36      0/25/25 peers     23 MiB db    5 MiB chain  0 bytes queue  227 KiB sync
2016-12-19 13:45:53  Imported #2839406 c8eb<E2><80><A6>e1f1 (27 txs, 0.61 Mgas, 15.59 ms, 3.67 KiB)
2016-12-19 13:46:05  Imported #2839407 bd33<E2><80><A6>fac1 (1 txs, 0.08 Mgas, 0.60 ms, 0.75 KiB)
2016-12-19 13:46:06      0/25/25 peers     23 MiB db    5 MiB chain  0 bytes queue  227 KiB sync
2016-12-19 13:46:36      0/25/25 peers     23 MiB db    5 MiB chain  0 bytes queue  227 KiB sync
2016-12-19 13:47:06      0/25/25 peers     23 MiB db    5 MiB chain  0 bytes queue  227 KiB sync
```
That last log line continues for another 10min until I restarted the node (during which no blocks & no other log lines appeared). Can attach full log if needed, but prior to this it ran fine since upgrading version.

Should I run `-l sync=trace` and/or any other flags to help diagnose?

**Comment by gavofyork** [comment]
@arkpar ?

**Comment by arkpar** [comment]
This has been fixed in 1.4.7 @russdirks @Gustav-Simonsson could you please give it anoter test with the latest version?

**Comment by russdirks** [comment]
@arkpar , I've been running 1.4.7 & 1.4.8 for about 3 days now and have not seen the node fall behind once, so it looks like the issue is fixed.
